Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/SparkJobHttpHandler.java

```diff
@@ -0,0 +1,82 @@
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.azure.hdinsight.spark.jobs;
+
+import com.microsoft.azure.hdinsight.common.JobViewManager;
+import com.microsoft.azure.hdinsight.sdk.common.HDIException;
+import com.microsoft.azure.hdinsight.sdk.rest.ObjectConvertUtils;
+import com.microsoft.azure.hdinsight.sdk.rest.spark.Application;
+import com.microsoft.azure.hdinsight.sdk.rest.spark.YarnAppWithJobs;
+import com.microsoft.azure.hdinsight.sdk.rest.spark.executor.Executor;
+import com.microsoft.azure.hdinsight.sdk.rest.spark.job.Job;
+import com.microsoft.azure.hdinsight.sdk.rest.spark.stage.Stage;
+import com.microsoft.azure.hdinsight.sdk.rest.spark.task.Task;
+import com.microsoft.azure.hdinsight.sdk.rest.yarn.rm.App;
+import com.microsoft.azure.hdinsight.spark.jobs.framework.JobRequestDetails;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+public class SparkJobHttpHandler implements HttpHandler {
+
+    @Override
+    public void handle(HttpExchange httpExchange) throws IOException {
+        httpExchange.getResponseHeaders().add("Access-Control-Allow-Origin", "*");
+        JobRequestDetails requestDetail = JobRequestDetails.getJobRequestDetail(httpExchange);
+        // TODO : get message from spark history server
+        try {
+            String path = requestDetail.getRequestPath();
+            if (path.equalsIgnoreCase("/applications/") && requestDetail.getAppId().equalsIgnoreCase("0")) {
+                List<Application> applications = JobViewCacheManager.getSparkApplications(requestDetail.getCluster());
+                Optional<String> responseString = ObjectConvertUtils.convertObjectToJsonString(applications);
+                JobUtils.setResponse(httpExchange, responseString.orElseThrow(IOException::new));
+            } else if (path.contains("application_graph")) {
```

Comment:
Suggest to use regex for these path matching. I suppose there are error logicals here, such as: "/tasks_summary/application_graph" will meet this condition, is it by design?